### PR TITLE
Add support for hard reboot to Beaker provision plugin

### PR DESCRIPTION
**Pull request by happz**
Pull Request Checklist

* [x] implement the feature
* [x] extend the test coverage

### CODE REVIEW

**Review by lukaszachy — APPROVED** [review]

**Review by psss — APPROVED** [review]
Thanks for implementing this! @happz, the checklist in the pull request description suggests that you still plan to add a test coverage?

**Comment by happz** [comment]
> Thanks for implementing this! @happz, the checklist in the pull request description suggests that you still plan to add a test coverage?

Test added in 7f621235.

File: tests/provision/reboot/test.sh [hunk 1]
```diff
@@ -27,6 +27,22 @@ rlJournalStart
         rlPhaseEnd
     fi
 
+    if [[ "$PROVISION_METHODS" =~ beaker ]]; then
+        rlPhaseStartTest "Beaker"
+            rlRun "tmt run --scratch -i $run provision -h beaker -i Fedora-39"
```
Comment (psss) [inline tests/provision/reboot/test.sh:32]: What about not specifying the image and going with the default one? We wouldn't have to update this regularly in the future.

Comment (happz) [reply tests/provision/reboot/test.sh:32]: That didn’t work for me with Beaker, I was gettin “bo such distro” errors :/

Comment (psss) [reply tests/provision/reboot/test.sh:32]: Hm, interesting, for me Beaker was unhappy with `Fedora-39` :-) What about `--image fedora`? Would that work for you?

Comment (happz) [reply tests/provision/reboot/test.sh:32]: For me? Sure. For Beaker? I don’t know, what you suggest is the default that did not work for me today :) Maybe I’m missing some piece of configuration, who knows. I’ll give it a try tomorrow - I didn’t pay too much attention to it today.

Comment (psss) [reply tests/provision/reboot/test.sh:32]: Strange. Could it be that your `/etc/tmt/mrack-provisioning-config.yaml` does not have effect? Or you have an old one? But still, even with the old mapping to `Fedora-37` it works for me... Don't know.

Comment (psss) [reply tests/provision/reboot/test.sh]: Agreed on chat to use the default image: c421c2b5

**Pull request merged**